src/pybind/mgr/mgr_module.py

```diff
@@ -1504,31 +1547,7 @@ def mon_command(self, cmd_dict: dict, inbuf: Optional[str] = None) -> Tuple[int,
 
         return r
 
-    def osd_command(self, cmd_dict: dict, inbuf: Optional[str] = None) -> Tuple[int, str, str]:
```

Comment:
@epuertat @Waadkh7 `osd_command` deletion is intended?

It was introduced in #42569.

I tested it and after executing:
```
ceph config set mgr mgr/telemetry/channel_perf true
ceph telemetry show-all
```
I get this error:
```
[telemetry ERROR root] 'Module' object has no attribute 'osd_command'
Traceback (most recent call last):
  File "/ceph/src/pybind/mgr/telemetry/module.py", line 376, in get_osd_histograms
    r, outb, outs = self.osd_command(cmd_dict)
AttributeError: 'Module' object has no attribute 'osd_command'
2021-11-09T14:48:40.951+0000 7f644d648700 -1 mgr handle_command module 'telemetry' command handler threw exception: local variable 'r' referenced before assignment
2021-11-09T14:48:40.951+0000 7f644d648700 -1 mgr.server reply reply (22) Invalid argument Traceback (most recent call last):
  File "/ceph/src/pybind/mgr/mgr_module.py", line 1640, in _handle_command
    return CLICommand.COMMANDS[cmd['prefix']].call(self, cmd, inbuf)
  File "/ceph/src/pybind/mgr/mgr_module.py", line 417, in call
    return self.func(mgr, **kwargs)
  File "/ceph/src/pybind/mgr/telemetry/module.py", line 1057, in show_all
    return 0, json.dumps(self.get_report('all'), indent=4, sort_keys=True), ''
  File "/ceph/src/pybind/mgr/telemetry/module.py", line 1067, in get_report
    return {'report': self.compile_report(channels=channels),
  File "/ceph/src/pybind/mgr/telemetry/module.py", line 899, in compile_report
    report['osd_perf_histograms'] = self.get_osd_histograms()
  File "/ceph/src/pybind/mgr/telemetry/module.py", line 380, in get_osd_histograms
    if r != 0:
UnboundLocalError: local variable 'r' referenced before assignment
```
@ljflores A good idea to detect these kind of impacts would be to add unit/integration tests to the telemetry module.

Reply:
@alfonsomthd no its not

Reply:
@alfonsomthd agreed, unit/integration tests for that function would be a good idea. I can take care of opening a PR for that soon. For context, this function is necessary in the Telemetry module to expose certain OSD commands, such as `ceph osd.* perf histogram dump`.

That said, I'm a little confused about why it's getting deleted here-- is there an issue with it?

Reply:
> That said, I'm a little confused about why it's getting deleted here-- is there an issue with it? @Waadkh7

@ljflores Waad answered above: it was accidental :stuck_out_tongue:

Reply:
The `osd_command` function is just like `mon_command`: https://github.com/ceph/ceph/blob/56463c4bc577d708a63680d13fb67840b379542a/src/pybind/mgr/mgr_module.py#L1482
The only difference is that `osd_command` is used to access commands in https://github.com/ceph/ceph/blob/master/src/osd/OSD.cc rather than https://github.com/ceph/ceph/blob/master/src/mon/MonCommands.h.

Reply:
> > That said, I'm a little confused about why it's getting deleted here-- is there an issue with it? @Waadkh7
> 
> @ljflores Waad answered above: it was accidental stuck_out_tongue

Oh I see! Sounds good. But I'm glad this came to light anyway because I do need to add a some tests for it.